codalab/rest/interpret.py

```diff
@@ -306,134 +314,86 @@ def resolve_interpreted_items(interpreted_items):
                 contents = interpret_genpath_table_contents(contents)
-                data = (header, contents)
-            elif mode == 'contents':
-                try:
-                    max_lines = int(properties.get('maxlines', DEFAULT_CONTENTS_MAX_LINES))
-                except ValueError:
-                    raise UsageError("maxlines must be integer")
 
-                target_info = rest_util.get_target_info(data, 0)
-                if target_info is not None and target_info['type'] == 'directory':
-                    data = ['<directory>']
+                item['rows'] = contents
+            elif mode == BlockModes.contents_block or mode == BlockModes.html_block or mode == BlockModes.image_block:
+                target_info = rest_util.get_target_info((item['bundles_spec']['bundle_infos'][0]['uuid'], item['target_genpath']), 0)
+
+                if target_info is not None and target_info['type'] == 'directory' and mode == BlockModes.contents_block:
+                    item['status']['code'] = 'ready'
+                    item['lines'] = ['<directory>']
                 elif target_info is not None and target_info['type'] == 'file':
-                    data = head_target(data, max_lines, replace_non_unicode=True)
-                else:
-                    data = None
-            elif mode == 'html':
-                target_info = rest_util.get_target_info(data, 0)
-                if target_info is not None and target_info['type'] == 'file':
-                    data = head_target(data, None)
+                    item['status']['code'] = 'ready'
+                    if mode == BlockModes.contents_block:
+                        item['lines'] = head_target((item['bundles_spec']['bundle_infos'][0]['uuid'], item['target_genpath']), item['max_lines'], replace_non_unicode=True)
+                    elif mode == BlockModes.html_block:
+                        item['html_lines'] = head_target((item['bundles_spec']['bundle_infos'][0]['uuid'], item['target_genpath']), item['max_lines'])
+                    elif mode == BlockModes.image_block:
+                        item['status']['code'] = 'ready'
+                        item['image_data'] = base64.b64encode(cat_target((item['bundles_spec']['bundle_infos'][0]['uuid'], item['target_genpath'])))
                 else:
-                    data = None
-            elif mode == 'image':
-                target_info = rest_util.get_target_info(data, 0)
-                if target_info is not None and target_info['type'] == 'file':
-                    data = base64.b64encode(cat_target(data))
-                else:
-                    data = None
-            elif mode == 'graph':
-                try:
-                    max_lines = int(properties.get('maxlines', DEFAULT_CONTENTS_MAX_LINES))
-                except ValueError:
-                    raise UsageError("maxlines must be integer")
-
+                    item['status']['code'] = 'not_found'
```

Comment:
probably should have constants defined for the various codes...

Reply:
yeah, sounds good.